backend/src/main/java/com/htecgroup/skynest/filter/CustomAuthenticationFilter.java

```diff
@@ -35,31 +36,34 @@ public class CustomAuthenticationFilter extends UsernamePasswordAuthenticationFi
   private final UserDetailsService userDetailsService;
   private final PasswordEncoder passwordEncoder;
 
+  private final UserService userService;
+
   private final ObjectMapper objectMapper;
 
   @Override
   public Authentication attemptAuthentication(
       HttpServletRequest request, HttpServletResponse response) throws AuthenticationException {
-    UserLoginRequest credentials = new UserLoginRequest();
 
+    UserLoginRequest credentials;
     try {
       credentials = objectMapper.readValue(request.getInputStream(), UserLoginRequest.class);
-
     } catch (IOException e) {
       log.error("Unable to authenticate, because of Input or Output error", e);
+      throw new UserException(UserExceptionType.INVALID_AUTHENTICATION_FORMAT);
     }
+
     UserDetails userDetails = userDetailsService.loadUserByUsername(credentials.getEmail());
-    log.info("The user: {} is trying to authenticate.", credentials.getEmail());
-    if (!credentials.getEmail().equals(userDetails.getUsername())) {
-      throw new UserException(UserExceptionType.INVALID_EMAIL_OR_PASSWORD);
-    }
+    log.info("The user: {} is trying to authenticate.", userDetails.getUsername());
+    if (!userService.isActive(userDetails.getUsername()))
+      throw new UserException(UserExceptionType.USER_NOT_ACTIVE);
 
-    if (!passwordEncoder.matches(credentials.getPassword(), userDetails.getPassword())) {
+    try {
```

Comment:
There is no need of second try, you can merge everything into one try with multiple catch clauses

Reply:
I've corrected it.